src/main/java/edu/byu/cs329/rd/ReachingDefinitionsBuilder.java

```diff
@@ -1,88 +1,232 @@
-    List<VariableDeclaration> parameterList = 
-        getParameterList(methodDeclaration.parameters());
-    Set<Definition> set = new HashSet<Definition>();
-
-    for (VariableDeclaration parameter : parameterList) {
-      Definition definition = createDefinition(parameter.getName(), null);
-      set.add(definition);  
-    }
-
-    return set;
-  }
-
-  private Definition createDefinition(SimpleName name, Statement statement) {
-    Definition definition = new Definition();
-    definition.name = name;
-    definition.statement = statement;
-    return definition;
-  }
-
-  private List<VariableDeclaration> getParameterList(Object list) {
-    @SuppressWarnings("unchecked")
-    List<VariableDeclaration> statementList = (List<VariableDeclaration>) (list);
-    return statementList;
-  }
-}
+package edu.byu.cs329.rd;
+
+import edu.byu.cs329.cfg.ControlFlowGraph;
+import edu.byu.cs329.rd.ReachingDefinitions.Definition;
+import edu.byu.cs329.utils.AstNodePropertiesUtils;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.eclipse.jdt.core.dom.Assignment;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.VariableDeclaration;
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+
+
+/**
+ * Builder for reaching definitions on a control flow graph.
+ */
+public class ReachingDefinitionsBuilder {
+  private List<ReachingDefinitions> rdList = null;
+  private Map<Statement, Set<Definition>> entrySetMap = null;
+  private Map<Statement, Set<Definition>> exitSetMap = null;
+  private Map<Statement, Set<Definition>> genSetMap = null;
+
+  /**
+   * Computes the reaching definitions for each control flow graph.
+   *
+   * @param cfgList the list of control flow graphs.
+   * @return the coresponding reaching definitions for each graph.
+   */
+  public List<ReachingDefinitions> build(List<ControlFlowGraph> cfgList) {
+    rdList = new ArrayList<ReachingDefinitions>();
+    for (ControlFlowGraph cfg : cfgList) {
+      ReachingDefinitions rd = computeReachingDefinitions(cfg);
+      rdList.add(rd);
+    }
+    return rdList;
+  }
+
+  private ReachingDefinitions computeReachingDefinitions(ControlFlowGraph cfg) {
+    exitSetMap = new HashMap<Statement, Set<Definition>>();
+    genSetMap = new HashMap<Statement, Set<Definition>>();
+    entrySetMap = new HashMap<Statement, Set<Definition>>();
+    Set<Definition> parameterDefinitions = createParameterDefinitions(cfg.getMethodDeclaration());
+    Statement start = cfg.getStart();
+    entrySetMap.put(start, parameterDefinitions);
+    
+    initialize(start, cfg);

```

Comment:
I like that you initialize gen sets beforehand here, that's how I solved the equals problem too. Nice that it is in a different function, makes it easy to read. Might elaborate on what you are initializing in the name